packages/app/src/app/pages/common/Modals/SelectWorkspaceToUpgrade/index.tsx

```diff
@@ -0,0 +1,171 @@
+import React from 'react';
+import {
+  Button,
+  Element,
+  Icon,
+  Select,
+  Stack,
+  Text,
+} from '@codesandbox/components';
+import track from '@codesandbox/common/lib/utils/analytics';
+
+import { useCreateCheckout } from 'app/hooks';
+import { useActions, useAppState } from 'app/overmind';
+import { SubscriptionType, TeamMemberAuthorization } from 'app/graphql/types';
+import { Alert } from '../Common/Alert';
+
+export const SelectWorkspaceToUpgrade: React.FC = () => {
+  const { dashboard, personalWorkspaceId, user } = useAppState();
+  const { openCreateTeamModal, modalClosed } = useActions();
+  const [checkout, createCheckout] = useCreateCheckout();
+  const [selectedTeam, setSelectedTeam] = React.useState(undefined);
+
+  const teamsToShow = dashboard.teams.filter(team => {
+    if (
+      team.id === personalWorkspaceId ||
+      team.subscription?.type === SubscriptionType.TeamPro
+    ) {
+      return false;
+    }
+
+    const teamAdmins = team.userAuthorizations
+      .filter(
+        ({ authorization }) => authorization === TeamMemberAuthorization.Admin
+      )
+      .map(({ userId }) => userId);
+
+    return teamAdmins.includes(user?.id);
+  });
+
+  React.useEffect(() => {
+    if (!selectedTeam) {
+      setSelectedTeam(teamsToShow[0]?.id);
+    }
+  }, [selectedTeam, teamsToShow]);
```

Comment:
If we move the `useState` down we can remove this `useEffect` and initialize the state with `teamsToShow[0]?.id`.

Reply:
Good catch! Removed here: [aa100a0](https://github.com/codesandbox/codesandbox-client/pull/7468/commits/aa100a0a9c6363c23771e29cc8e4a970cdb5a5d8). This was a leftover because at first I was searching for the data and using the effect to react to when it was loaded, but this data already exists when we open the modal, so the fetch wasn't needed.